networkx/algorithms/centrality/second_order_centrality.py

```diff
@@ -106,6 +109,20 @@ def second_order_centrality(G):
         raise nx.NetworkXException("Empty graph.")
     if not nx.is_connected(G):
         raise nx.NetworkXException("Non connected graph.")
+    if [True for (i, j, d) in G.edges(data=True)
+            if 'weight' in d and d['weight'] < 0]:
+        raise nx.NetworkXException("Graph has negative edge weights.")
+
+    def unbias_graph(G):
+        G = nx.DiGraph(G)
+        d_max = np.max([G.in_degree(i,
```

Comment:
Nodes don't have to be integers...and ```in_degree```returns a dict of all nodes if no node is specified.
So the code is probably faster and more readable something like this:

    in_deg = G.in_degree(weight='weight')
    d_max = np.max(in_deg.values())
    for n, deg in d_max.items():
        if deg < d_max:
            G.add_edge(n, n, weight=d_max-deg)